encoder.lua

```diff
@@ -67,74 +67,152 @@ local function toms(timestamp)
     return string.format("%.3f", timestamp)
 end
 
-local function quality_to_crf(quality, max_crf)
-    -- Quality is from 0 to 100. (for avif images) CRF is from 0 to 63 and reversed.
-    return math.floor((100 - quality) / 100 * max_crf)
+local function rescale_quality(quality, min_q, max_q)
+    local scaled = min_q + (max_q - min_q) * quality / 100
+    -- Round to the nearest integer that's better in quality.
+    if min_q > max_q then
+        return math.floor(scaled)
+    end
+    return math.ceil(scaled)
```

Comment:
This function was previously named quality_to_crf but with the new name it's hard to understand what it does.

Reply:
Renamed to `fit_quality_percentage_to_range`. JPEG function was also removed since `qscale` is not CRF: https://stackoverflow.com/a/40673522. AVIF and JPEF constants were moved inside their respective functions.